util/git/client.go

```diff
@@ -527,6 +565,21 @@ func (m *nativeGitClient) runCredentialedCmd(command string, args ...string) err
 	return err
 }
 
+// runCredentialedCmd is a convenience function to run a git command with username/password credentials
+// nolint:unparam
+func (m *nativeGitClient) runCredentialedCmdWithOutput(command string, args ...string) (string, error) {
```

Comment:
Rather than adding a new `runCredentialedCmdWithOutput`, I recommend instead just changing the return value of `runCredentialeldCmd` to return `(string, error)` and updating all the callers  (unless there was a reason not to do so)